### PR TITLE
feat: syncmap for channels

### LINKED ISSUE #54: allow config to change concurrently
Commands and otherwise may want to update channel configs. Make the channels map synchronized so we can.

### PR DESCRIPTION
**Pull request by Karitham**
fixes #54

### CODE REVIEW

File: syncmap/syncmap.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+// Len returns the number of elements in the map.
+func (m *Map[K, V]) Len() int {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return len(m.m)
+}
+
+// Range calls f sequentially for each key and value present in the map.
```
Comment (zephyrtronium) [inline syncmap/syncmap.go:50]: wrong doc

File: syncmap/syncmap.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// If f returns false, range stops the iteration.
+func (m *Map[K, V]) All() iter.Seq2[K, V] {
```
Comment (zephyrtronium) [inline syncmap/syncmap.go:52]:
```go
m.mu.Lock()
defer m.mu.Unlock()
return maps.All(maps.Clone(m.m))

Comment (Karitham) [reply syncmap/syncmap.go:51]: did something else that should still be correct. I've tried using testing/quick to test it but I don't think my usage of it is conclusive.

File: .envrc
Comment (zephyrtronium) [inline .envrc:1]: ??

File: flake.lock
Comment (zephyrtronium) [inline flake.lock:1]: ??

File: flake.nix
Comment (zephyrtronium) [inline flake.nix:1]: ??

File: syncmap/syncmap_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		t.Errorf("Expected early termination after 2 elements, got %d", count)
+	}
+
+	// Test concurrent modification
+	go func() {
+		m.Store("four", 4)
+		m.Delete("two")
+	}()
+
+	time.Sleep(10 * time.Millisecond)
```
Comment (zephyrtronium) [inline syncmap/syncmap_test.go:99]: Sleeping is not synchronizing

**Review by zephyrtronium — CHANGES_REQUESTED** [review]

File: flake.nix
Comment (Karitham) [reply flake.nix:1]: oops

File: syncmap/syncmap.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+// Len returns the number of elements in the map.
+func (m *Map[K, V]) Len() int {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return len(m.m)
+}
+
+// All iterates over all elements in the map
+func (m *Map[K, V]) All(yield func(K, V) bool) {
```
Comment (zephyrtronium) [inline syncmap/syncmap.go:50]: The convention is that All() always returns an iter.Seq/2, rather than being the iterator itself, for consistency.

**Review by zephyrtronium — CHANGES_REQUESTED** [review]

**Pull request merged**